### ISSUE TITLE
Superfluous newlines

### ISSUE THREAD

**Issue by finkf** [OCR-D/ocrd_tesserocr#18]
At the moment, superfluous newlines are appended to the `TextEquiv/Unicode` entries:
```xml
                    <pc:TextEquiv>
                        <pc:Unicode>Groſzmaͤchtigſter</pc:Unicode>
                    </pc:TextEquiv>
                    <pc:TextEquiv>
                        <pc:Unicode>stzmächtigstcr
</pc:Unicode>
```

**Comment by bertsky** [comment]
I also noticed this but was not sure whether this is deliberate and correct. It can be easily changed by calling `tessapi.SetLineSeparator("")` right after `tessapi.SetRectangle()` for lines, and `tessapi.SetParagraphSeparator("")` for regions, respectively.

**Comment by kba** [comment]
Thanks for raising and providing a possible solution.

There should not be any separators for either line or block. I was afraid this was because of XML pretty-printing which is trickier to get right, I'd be hesitant to just chomp trailing newlines of textnodes. But if it's just a matter of adapting tesseract API calls, seems straightforward to fix.

@finkf Have you tried the #17 version? Ideally, we should merge that one first and fix this one in the merged code.

**Comment by bertsky** [comment]
It seems there used to be a `TextMergeRuleSimpleType` (featuring values like `add-whitespace` and `add-line-break`) which is now commented in pagecontent.xsd. It is not referenced in any of the update documentation on github, though.

**Comment by bertsky** [comment]
@kba I would also recommend pulling #17 first, since it would apply on both levels.

**Comment by finkf** [comment]
OK. I can wait till #17 is merged.

**Comment by bertsky** [comment]
It turns out that `SetLineSeparator` and `SetParagraphSeparator` are only available for `ResultIterator` – which we never use above the word level. 

So we have:
- `SetRectangle()` gives trailing newlines by default – at _all_ levels (including Word).
- `GetIterator()` does not give trailing newlines up to the Word level, and can be controlled by aforementioned functions above that.
Very strange indeed!

Therefore I just added `strip()` to all the non-iterator results.

**Comment by finkf** [comment]
`strip()` seems fine to me.

**Comment by kba** [comment]
Perhaps strip("\n") so as to only strip newlines?

**Comment by bertsky** [comment]
Oh right, good point! (And perhaps also "\f" form feed, the new page separator.)

**Comment by finkf** [comment]
Do we really need to care about leading/trailing whitespace? I would prefer to have `"word"` instead of possible `" word \t"` problems, even if the OCR should emit such things. If we let such things happen I think that everyone will do a `strip()` on every line, word (or glyph) text.

**Comment by kba** [comment]
I was thinking of https://github.com/OCR-D/spec/issues/72 where whitespace would be significant. True, `" word \t"` would be annoying but if that is what the engine thinks is in the text, I would pass it on.

Compromise: Make it parameterizable to strip all whitespace and default to just right-strip NL and FF?

**Comment by finkf** [comment]
@kba you have a valid point there and I do not like the compromise (because of the added complexity), so I vote for `strip("\n\f")` (or whatever the right syntax for stripping `"\t"` and `"\f"` is).

**Comment by finkf** [comment]
Whats about `"\r"`?

**Comment by kba** [comment]
From (not exhaustively) scanning [baseapi.cpp](https://github.com/tesseract-ocr/tesseract/blob/master/src/api/baseapi.cpp#L1971), I think tesseract only inserts `\n`.

**Comment by kba** [comment]
Implemented in d459ae0
